src/Context/PostsContext.tsx

```diff
@@ -0,0 +1,196 @@
+import {
+  ReactNode, createContext, useCallback, useContext, useEffect, useState,
+} from 'react';
+import { Post } from '../types/Post';
+import { client } from '../utils/fetchClient';
+import { User } from '../types/User';
+import { Comment } from '../types/Comment';
+
+interface PostsContextInterface {
+  posts: Post[] | null;
+  users: User[];
+  loader: boolean;
+  comments: Comment[];
+  selectedUserId: number;
+  handleClickOnUsers: (id: number) => void;
+  isActive: boolean;
+  setIsActive: (active: boolean) => void;
+  error: boolean;
+  handleClickOpenComments: (id: number) => void;
+  isOpenPost: boolean;
+  chosenPost: number,
+  loadingPost: boolean,
+  postError: boolean,
+  handleDeleteComment: (id: number) => void;
+  isWriteComment: boolean;
+  setIsWriteComment: (isWriteComment: boolean) => void;
+  AddComment: (newComment: {}) => void;
+  loadAdd: boolean;
+}
+
+const PostsContext = createContext<PostsContextInterface>({
+  posts: [],
+  loader: false,
+  users: [],
+  comments: [],
+  selectedUserId: 0,
+  handleClickOnUsers: () => { },
+  isActive: false,
+  setIsActive: () => { },
+  error: false,
+  handleClickOpenComments: () => { },
+  isOpenPost: false,
+  chosenPost: 0,
+  loadingPost: false,
+  postError: false,
+  handleDeleteComment: () => { },
+  isWriteComment: false,
+  setIsWriteComment: () => { },
+  AddComment: () => { },
+  loadAdd: false,
+});
+
+export const PostsContextProvider = ({ children }: { children: ReactNode }) => {
+  const [isWriteComment, setIsWriteComment] = useState(false);
+  const [posts, setPosts] = useState<Post[] | null>(null);
+  const [loader, setLoader] = useState(false);
+  const [users, setUsers] = useState<User[]>([]);
+  const [comments, setComments] = useState<Comment[]>([]);
+  const [selectedUserId, setSelectedUserId] = useState(0);
+  const [isActive, setIsActive] = useState(false);
+  const [error, setError] = useState(false);
+  const [chosenPost, setChosenPost] = useState(0);
+  const [isOpenPost, setIsOpenPost] = useState(false);
+  const [loadingPost, setLoadingPost] = useState(false);
+  const [postError, setPostError] = useState(false);
+  const [changes, setChanges] = useState(false);
+  const [loadAdd, setLoadAdd] = useState(false);
+
+  useEffect(() => {
+    const url = '/posts';
+
+    setLoader(true);
+    const loadData = async () => {
+      try {
+        const data = await client.get<Post[]>(url);
+
+        setPosts(data.filter(post => post.userId === selectedUserId));
+      } catch {
+        setError(true);
```

Comment:
would be better so manage error in finally block

Reply:
why? I try catch errors.  I need this here: https://prnt.sc/buS8I51QT8Vm